docs/dev-overview.md

```diff
@@ -0,0 +1,17 @@
+# Development environment options
+
+Zulip offers a wide range of options for how to install the
+development environment:
+
+* [Detailed tutorial for Vagrant development environment](dev-env-first-time-contributors.html)
```

Comment:
I suggest noting somewhere near here that the Vagrant setup is the recommended method for first-time contributors, if they don't know where else to start.